corehq/apps/app_manager/management/commands/fix_app_docs_with_empty_keys.py

```diff
@@ -0,0 +1,248 @@
+                            current[last_part][''] = value
+                    else:
+                        # Empty key at root level
+                        app_json[''] = value
+
+                Application.wrap(app_json).save()
+                restored_apps += 1
+                self.stdout.write(
+                    self.style.SUCCESS(f"Restored empty keys for app: {app_id} ({processed_apps}/{total_apps})")
+                )
+            except Exception as e:
+                self.stdout.write(self.style.ERROR(f"Error restoring app {app_id}: {e}"))
+
+        self.stdout.write(
+            self.style.SUCCESS(f"Rollback completed. Restored {restored_apps} out of {total_apps} apps.")
+        )
+
+    def find_docs_with_empty_keys(self):
+        # Query Elasticsearch to find documents with empty keys
+        ids_with_empty_keys = []
+        query = {
+            "query": {
+                "match_all": {}
+            }
+        }
+        # Count total documents
+        total_docs = app_adapter.count(query)
+        processed_docs = 0
+        scroll = app_adapter.scroll(query, size=1000)
+        for doc in scroll:
+            processed_docs += 1
+            if self.has_empty_keys(doc):
+                ids_with_empty_keys.append(doc['_id'])
+            # Print progress every 1000 documents
+            if processed_docs % 1000 == 0:
+                self.stdout.write(f"Processed {processed_docs}/{total_docs} documents...")
+        self.stdout.write(
+            self.style.SUCCESS(f"Finished processing. Total documents with empty keys: {len(ids_with_empty_keys)}")
+        )
+        return ids_with_empty_keys
+
+    def write_ids_to_file(self, ids):
+        with open(self.ids_file, 'w') as f:
+            json.dump(ids, f)
+        self.stdout.write(self.style.SUCCESS(f"Wrote {len(ids)} document IDs to {self.ids_file}"))
+
+    def fix_documents(self, ids):
+        total_docs = len(ids)
+        fixed_docs = 0
+
+        for i, doc_id in enumerate(ids):
+            try:
+                app = Application.get(doc_id)
+                app_json = self.remove_empty_keys(app)
+                Application.wrap(app_json).save()
+                fixed_docs += 1
+                self.stdout.write(
+                    self.style.SUCCESS(f"Fixed document with ID: {doc_id} ({i+1}/{total_docs})")
+                )
+            except Exception as e:
+                self.stdout.write(self.style.ERROR(f"Error processing document with ID {doc_id}: {e}"))
+
+        self.stdout.write(
+            self.style.SUCCESS(
+                f"All documents have been processed. Fixed {fixed_docs} out of {total_docs} documents."
+            )
+        )
+
+    def log_deleted_keys(self, app_id, path, value):
+        with open(self.log_file, 'a') as log_file:
+            log_entry = {
+                'app_id': app_id,
+                'path': path,
+                'value': json.dumps(value)
```

Comment:
Why is `value` JSON-encoded twice (here and below at `json.dump(log_entry, log_file)`)?
```suggestion
                'value': value,
```

Reply:
Yeah. You are correct. I don't know why I was thinking that I need to do it twice 🤔

Reply:
Ohh! i remember I had this thought that maybe newlines in values can cause issues in newline seperated json but I just tested it locally and and things were working fine. So I will update it as per you suggestion.

Reply:
https://github.com/dimagi/commcare-hq/pull/35889/commits/2ba5e0a26df2f978b388c249ec003828d0384612